tests/unit/experiments/providers/simple_experiment_tests.py

```diff
@@ -103,6 +103,33 @@ def get_targeting_config():
     }
     return targeting_cfg
 
+def get_simple_override_config():
+    override_config = [
+        {
+            'override_variant_1':{'EQ': {'field':'user_id', 'value':'t2_1'}}
+        },
+        {
+            'override_variant_2':{'EQ': {'field':'user_id', 'value':'t2_2'}}
+        },
+        {
+            'override_variant_3':{'EQ': {'field':'user_id', 'values':['t2_2', 't2_3']}},
+        },
+        {
+            'override_variant_1':{'EQ': {'field':'user_id', 'values':['t2_1', 't2_4']}},
+        }
+    ]
+
+    return override_config
+
+def get_dict_override_config():
+    override_config = {
+        'override_variant_1':{'EQ': {'field':'user_id', 'value':'t2_1'}},
+        'override_variant_2':{'EQ': {'field':'user_id', 'value':'t2_2'}},
+        'override_variant_3':{'EQ': {'field':'user_id', 'values':['t2_2', 't2_3']}},
```

Comment:
I feel like we shouldn't allow duplicate `key=value` overrides, that sounds like it would get very confusing when someone thinks they added an override but aren't seeing it show up because they have another override earlier in the config that overrides their new one.  I'm also not sure what the benefit of this is?  In this case, `user_id=t2_2` is always going to return `override_variant_2`, so why even let them set it as another override value?

Reply:
This example is meant to be pretty trivial; think of it instead as the override target contains a set of things based on the input parameters. In this case, the sets are "user_id=t2_2" and "user_id in [t2_2, t2_3]", but the actual sets provided can be largely arbitrary. It could be "user_id is t2_2" and "user is moderator". Determining the overlap there is much harder.
It could even be "user is supertype" and "user is subtype"; without having the type context calculating the overlapping segment is impossible.
I don't think it's worth including a bunch of complicated code to try to calculate overlapping sets here when allowing the "fall through" behaviour is cleaner, simpler, and not application-specific.
_If_ a client wants to disallow overlapping segments, it should be done when creating the configuration with the contextual knowledge of whatever user sets the client has.

Reply:
> I don't think it's worth including a bunch of complicated code to try to calculate overlapping sets

That's fair, I didn't consider that